rov/movement/mapper/Complex_1.py

```diff
@@ -0,0 +1,275 @@
+from numpy import linalg
+import numpy as np
+import pprint as pp
+
+
+class Complex():
+    """
+    This code is a rewrite of MutatorMatrix to simplify and clarify the code without removing the source
+    This thrust-mapping works by solving the least squared solution of a system determined by the thrusters locations
+    and the directions of the thrusters.
+    The system is the 6x8 matrix made of columns of each thrusters affect on each component X, Y, Z. ROLL, PITCH, YAW
+    multiplied by the 8x1 thrust map matrix (which we are solving for) equal to the 6x1 desired thrust matrix
+    To find the least square solution we find the pseudo-inverse of the 6x8 matrix (A) and multiply both sides of the
+    equation by it. If a inverse of the matrix A exists the pseudo-inverse(A) = inverse(A) if not then
+    pseudo-inverse(A) * A can be ignored because math leaving
+    thrust map matrix = pseudo-inverse(A) * desired thrust
+    *For location and rotation vectors, the first four thrusters are the horizontal thrusters: first is front left, 
+    second is front right, third is back left, and fourth is back right. The last four are the vertical thrusters in
+    the same order: front left, front right, back left, and back right.

```

Comment:
Explain at the end, how outside classes should use this class. What functions to call to do what, when, etc.